lib/internal/readline/interface.js

```diff
@@ -1324,41 +1321,15 @@ class Interface extends InterfaceConstructor {
    * @returns {InterfaceAsyncIterator}
    */
   [SymbolAsyncIterator]() {
-    if (this[kLineObjectStream] === undefined) {
-      if (Readable === undefined) {
-        Readable = require('stream').Readable;
-      }
-      const readable = new Readable({
-        objectMode: true,
-        read: () => {
-          this.resume();
-        },
-        destroy: (err, cb) => {
-          this.off('line', lineListener);
-          this.off('close', closeListener);
-          this.close();
-          cb(err);
-        },
-      });
-      const lineListener = (input) => {
-        if (!readable.push(input)) {
-          // TODO(rexagod): drain to resume flow
-          this.pause();
-        }
-      };
-      const closeListener = () => {
-        readable.push(null);
-      };
-      const errorListener = (err) => {
-        readable.destroy(err);
-      };
-      this.on('error', errorListener);
-      this.on('line', lineListener);
-      this.on('close', closeListener);
-      this[kLineObjectStream] = readable;
+    if (!this[kLineObjectStream]) {
+      this[kLineObjectStream] = EventEmitter.on(
+        this, 'line', {
+          close: ['close'],
+          pauseThreshold: 1024,
+          firstEventParam: true,
```

Comment:
Since these params won't be exposed to the public can you please use symbols for them? e.g.

```
const kFirstEventParam = Symbol('nodejs.kFirstEventParam');
```

And then pass that instead:

```
  [kFirstEventParam]: true
```

Since these aren't exposed to users?

Reply:
I can do that.
But maybe it's a good thing to expose it to users, it can be useful in other situations too.
Regardless, it can be discussed in a further moment, so I'll change it to a internal symbol

Reply:
@benjamingr the problem here is that I'm using the firstEventParam on the interface.js, so, even if I create an internal symbol into events.js, I would need to export it. Is there some special place you folks use to put internal symbols shared between files?

Reply:
@Farenheith the thing is we are touching a public API here (on) without a lot of feedback and since you've already been working on this PR for a while I don't want to endanger delaying it further so I would rather not expose any public API changes to `on` here and do it in a follow up PR.

Reply:
@benjamingr okay, I created a new file to keep this symbol inside the internal folder. I think it's enough to attend that.
Please take a look if it solves the issue and if there's anything else to fix, just let me know!